api/preprints/views.py

```diff
@@ -69,6 +69,30 @@
 from osf.metrics import PreprintDownload, PreprintView
 
 
+class RejectPreprintEditMixin:
+    '''Override method to reject modify requests for old preprint versions (except for withdrawal)'''
+    def update(self, request, *args, **kwargs):
+        preprint = self.get_preprint(check_object_permissions=False)
+        if preprint.is_latest_version or preprint.is_retracted:
```

Comment:
So maybe don't need to care whether 1preprint.is_retracted1 or not, it should have been handled but super or other  mix-ins.